frontend/document/koptinterface.lua

```diff
@@ -177,6 +177,13 @@ function KoptInterface:getContextHash(doc, pageno, bbox, hash_list)
     table.insert(hash_list, bbox.y1)
     table.insert(hash_list, canvas_size.w)
     table.insert(hash_list, canvas_size.h)
+    -- NOTE: only add an entry for the rendering
+    -- mode if colored output is enabled to avoid
+    -- invalidating previously cached data on B&W
+    -- devices.
```

Comment:
(I don't like long lines for comments, but neither too short lines :) Also, this comment feels like short term - just saying there's no migration need now that we introduce color, dunno if it's worth keeping it.)

Reply:
Why short term?

Reply:
I guess you're thinking about color devices that will get color kopt when updating to a new koreader, and you don't want to reuse the cached b&w stuff from the previous version.
After that, users will only get cached color stuff. So "avoid invalidating previously cached data" won't really happen after that.
(I know they can switch color <=> B&W after that, but then, it's the same for all other bits we use in hash, we don't add a comment for each of them, it is somehow obvious.)

"not invalidating" it is not really what matters :) we could just trash it. It's being sure to not re-use something that does not correspond to the current settings that matters.

Reply:
We don't even trash it anyway. It might still be there if the user switches back to black and white. It might not. In any case it's better now. ;-)